recipes/imagemagick/all/conanfile.py

```diff
@@ -0,0 +1,399 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, MSBuild
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+
+
+class ImageMagicConan(ConanFile):
+    name = "imagemagick"
+    description = ("ImageMagick is a free and open-source software suite for displaying, converting, and editing "
+                   "raster image and vector image files")
+    topics = ("conan", "imagemagick", "images", "manipulating")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://imagemagick.org"
+    license = "ImageMagick"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "pkg_config"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "hdri": [True, False],
+               "quantum_depth": [8, 16, 32],
+               "with_zlib": [True, False],
+               "with_bzlib": [True, False],
+               "with_lzma": [True, False],
+               "with_lcms": [True, False],
+               "with_openexr": [True, False],
+               "with_jpeg": [None, "libjpeg", "libjpeg-turbo"],
+               "with_openjp2": [True, False],
+               "with_pango": [True, False],
+               "with_png": [True, False],
+               "with_tiff": [True, False],
+               "with_webp": [True, False],
+               "with_xml2": [True, False],
+               "with_freetype": [True, False],
+               "with_djvu": [True, False],
+               "utilities": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "hdri": True,
+                       "quantum_depth": 16,
+                       "with_zlib": True,
+                       "with_bzlib": True,
+                       "with_lzma": True,
+                       "with_lcms": True,
+                       "with_openexr": True,
+                       "with_jpeg": "libjpeg",
+                       "with_openjp2": True,
+                       "with_pango": True,
+                       "with_png": True,
+                       "with_tiff": True,
+                       "with_webp": False,
+                       "with_xml2": True,
+                       "with_freetype": True,
+                       "with_djvu": False,
+                       "utilities": True}
+    exports_sources = "patches/*"
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "ImageMagick"  # name is important, VisualMagick uses relative paths to it
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _is_mingw_windows(self):
```

Comment:
This property is only needed to set `win_bash` of `AutotoolsBuildEnvironment`.
I think this method can be removed and the `AutotoolsBuildEnvironment` construction replaced with:
```python
autotools = AutotoolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
```
When using autotools on Windows, winbash is always needed independent of the compiler.